eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -88,16 +88,13 @@ jobs:
         inputs:
           packageType: sdk
           version: "$(DotNetCoreSDKVersion)"
-      - script: "dotnet test eng/service.proj --filter TestCategory!=Live --logger trx /p:ServiceDirectory=${{parameters.ServiceDirectory}}"
-        displayName: "Build & Test"
-        env:
-          DOTNET_SKIP_FIRST_TIME_EXPERIENCE: 1
-          DOTNET_CLI_TELEMETRY_OPTOUT: 1
-          DOTNET_MULTILEVEL_LOOKUP: 0
-      - task: PublishTestResults@2
-        condition: succeededOrFailed()
-        inputs:
-          testResultsFiles: "**/*.trx"
-          testRunTitle: "$(OSName) DotNet $(DotNetCoreRuntimeVersion)"
-          testResultsFormat: "VSTest"
-          mergeTestResults: true
+      - template: ../steps/test-and-publish.yml
```

Comment:
Realizing that this comment is coming in a bit late: Why not have `TFM` be a parameter of the matrix entries? The matrix would be more repetitive, but we could avoid the conditionals and having to break out `test-and-publish.yml`

```
        Linux:
          OSName: "Linux"
          OSVmImage: "ubuntu-16.04"
          TFM: netcoreapp2.1
        Windows_NetCoreApp:
          OSName: "Windows"
          OSVmImage: "vs2017-win2016"
          TFM: netcoreapp2.1
        Windows_NetFramework:
          OSName: "Windows"
          OSVmImage: "vs2017-win2016"
          TFM: net461
        MacOs:
          OSName: "MacOS"
          OSVmImage: "macOS-10.13"
          TFM: netcoreapp2.1
```